BondarenkoBE/Вектор.cpp

```diff
@@ -0,0 +1,165 @@
+﻿#include "pch.h"
+#include <iostream>
+#include <string>
+using T = int;
+
+class Vector{
+	public:
+		Vector(unsigned size, const T& val);
+		unsigned size();
+		void resize(unsigned size);
+		bool insert(unsigned pos, const T& val);
+		void push_back(const T& val);
+		bool contains(const T& val);
+		T& operator [](unsigned index);
+		int find(const T& val);
+		T* data();
+		Vector(const Vector&vec);
+		Vector& operator =(const Vector&vec);
+		~Vector();
+	private:
+		T* Data = nullptr;
+		unsigned Size = 0;
+};
+
+Vector::Vector(unsigned size = 1, const T& val = 0) {
+	Data = new T[size];
+	Size = size;
+	for (unsigned i = 0; i < Size; ++i)
+		Data[i] = val;
+}
+
+unsigned Vector::size() {
+	return Size;
+}
+
+void Vector::resize(unsigned size) {
+	T* Buf_Data = new T[size];
+	if (size > Size) {
+		memcpy(Buf_Data, Data, Size * sizeof(T));
+		for (unsigned i = Size; i < size; ++i)
+			Buf_Data[i] = 0;
+	}
+	else memcpy(Buf_Data, Data, size * sizeof(T));
+	delete []Data;
+	Data = Buf_Data;
+	Size = size;
+}
+
+bool Vector::insert(unsigned pos, const T& val) {
+	if (pos > Size) return false;
+	resize(Size + 1);
+	memmove(Data + pos + 1, Data + pos, (Size - pos - 1) * sizeof(T));
+	Data[pos] = val;
+	if (Data[pos] == val)
+		return true; 
+	else
+		return false;
+}
+
+void Vector::push_back(const T& val) {
+	resize(Size + 1);
+	Data[Size - 1] = val;
+}
+
+bool Vector::contains(const T& val) {
+	for (unsigned i = 0; i < Size; ++i)
+		if (Data[i] == val) return true;
+	return false;
+}
+
+T& Vector::operator [](unsigned index) {
+	return Data[index];
+}
+
+int Vector::find(const T& val) {
+	for (unsigned i = 0; i < Size; ++i)
+		if (Data[i] == val) return i;
+	return -1;
+}
+
+T* Vector::data() {
+	return Data;
+}
+
+Vector::Vector(const Vector&vec) {
+	Size = vec.Size;
+	delete []Data;
```

Comment:
Possibly dangerous place
As it is constructor, first of all, we can assume that no allocations were called